documentation/git_training.md

```diff
@@ -0,0 +1,258 @@
+
+*Finding your current branch*
+
+If you are not sure what branch you are working on git branch will tell you what branch you are currently on. It's always a good idea to run git branch before pushing. The code below should highlight in green the branch you are currently on.
+
+```{bash}
+git branch
+```
+
+We created an issue in the economic mobility repository that we will use as a practice for the remaining part of this training. It is issue number 228. In your economic mobility folders lets git bash and create this branch.
+
+```{bash}
+git checkout -b iss228
+```
+
+## Updating and Pushing Your Work
+
+Now that we are on our practice branch, we are going to walk through the Git workflow.
+
+The most common Git commands you will use while working on updates are *git status, git add, git commit, and git push*. These four commands will be 95% of your git workflow. These tell Git which files to track, when to take snapshots of those files, and when to send those snapshots to GitHub for others to see.
+
+Once you have made updates to your code and it is ready for review you will go through the following steps. For the sake of practice, let's open a text file and save it to the main folder. Name it personalized to you "test_file\_\<your_name\>" and put whatever you want in it (maybe a little note saying how excited you are for this work).
+
+A helpful function git status prints out important information about your repo. Use it before and after most commands to understand how files have changed in your repo.
+
+```{bash}
+git status
+```
+
+We can see that the added text file is shown in red text. This tells us it is a change from how we cloned the repo and has not yet been prepared to be merged with our main repo.
+
+#### Adding and Committing
+
+To prepare our new file for the main repo we use git add which adds a file to Git's staging area. It tells Git, "Hey, look at this file." After adding the text file, if we check git status again the test_file.txt will now appear in green.
+
+```{bash}
+git add test_file.txt
+git status
+```
+
+Now that we have staged our file, we are ready to commit it. The command git commits commits changes made to added files to the repository. It says, "Hey, take a snapshot of the files I've added."
+
+```{bash}
+git commit -m "<message>"
+```
+
+Note: sometimes you may receive the message:
+
+type :quit <Enter> to quit VIM
+
+Please follow this [link](https://stackoverflow.com/questions/11828270/how-do-i-exit-vim) to see how you can resolve this issue.
+
+#### Pushing Changes
+
+We have now committed the changes and it's time to send them to our main repository. The command git push pushes your local commits to the remote repository on GitHub. It says, "hey, push all my snapshots to a shared folder so that other people can see them." The command below will send your changes to the GitHub page under a new branch but will not immediately change the main repo.
```

Comment:
Double-check that the user is going to push to main. I thought maybe Aaron said that this should be something you and he are responsible to do? Or at least that you'd need to give clear authorization/instruction to do so. 

Oh, never mind! It looks like this won't go to main really ;). This is why I need this training ;)